src/ebay-tri-state-checkbox/tri-state-checkbox.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { ChangeEvent, ComponentProps, FC, FocusEvent, KeyboardEvent, useState } from 'react'
+import classNames from 'classnames'
+import { EbayIcon } from '../ebay-icon'
+import { EbayChangeEventHandler, EbayFocusEventHandler, EbayKeyboardEventHandler } from '../common/event-utils/types'
+
+type Size = 'default' | 'large'
+type CheckboxState = 'true' | 'false' | 'mixed'
+type InputProps = Omit<ComponentProps<'input'>, 'size' | 'onChange' | 'onFocus' | 'onKeyDown'>
+type EbayTriStateCheckboxProps = {
+    checked?: CheckboxState,
+    defaultChecked?: CheckboxState,
+    skipMixed?: boolean,
+    size?: Size;
+    onChange?: EbayChangeEventHandler<HTMLInputElement, { value: string | number, checked: CheckboxState }>;
+    onFocus?: EbayFocusEventHandler<HTMLInputElement, { value: string | number, checked: CheckboxState }>;
+    onKeyDown?: EbayKeyboardEventHandler<HTMLInputElement, { value: string | number, checked: CheckboxState }>;
+    inputRef?: React.LegacyRef<HTMLInputElement>;
+}
+
+const EbayTriStateCheckbox: FC<InputProps & EbayTriStateCheckboxProps> = ({
+    id,
+    size = 'default',
+    className,
+    style,
+    checked,
+    defaultChecked = 'false' as CheckboxState,
+    skipMixed = false,
+    onChange = () => {},
+    onFocus = () => {},
+    onKeyDown = () => {},
+    children,
+    inputRef,
+    ...rest
+}) => {
+    const [checkboxState, setCheckboxState] = useState<CheckboxState>(checked || defaultChecked)
```

Comment:
We use only `defaultChecked` as the initial state. We don't use the state when it is controlled. If we want the to always use the state, we would need to add an `useEffect` to update the state as the `checked` value changes but it is not ideal (as mentioned in this react team [post](https://react.dev/learn/you-might-not-need-an-effect))

You can check the `EbayCheckbox` for reference as this is also handle [here](https://github.com/eBay/ebayui-core-react/blob/main/src/ebay-checkbox/checkbox.tsx#L34)